src/main/java/com/cag/cagbackendapi/error/ErrorDetails.java

```diff
@@ -0,0 +1,33 @@
+package com.cag.cagbackendapi.error;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Date;
+
+public class ErrorDetails {
+    private final Date time;
+    private final String restErrorMessage;
+    private final String detailedMessage;
+
+    public Date getTime() {
+        return time;
+    }
+
+    public String getRestErrorMessage() {
+        return restErrorMessage;
+    }
+
+    public String getDetailedMessage() {
+        return detailedMessage;
+    }
+
+    @JsonCreator
+    public ErrorDetails(@JsonProperty("time") Date time,
```

Comment:
Why does this error message also give the time?

Reply:
Because when you get an error, it's often one of many. In order to make it really easy to determine what order errors happened in, it's always a good practice to add a timestamp to errors in any application you build. Gives you more clues about what went wrong!